tests/unit/test.program.js

```diff
@@ -480,21 +507,154 @@ describe('program.main', () => {
       return configObject;
     });
 
-    return execProgram(
+    await execProgram(
       ['lint', '--config', 'path/to/web-ext-config.js'],
       {
         commands: fakeCommands,
         runOptions: {
           loadJSConfigFile: fakeLoadJSConfigFile,
         },
-      })
-      .then(() => {
-        const options = fakeCommands.lint.firstCall.args[0];
-        // This makes sure that the config object was applied
-        // to the lint command options.
-        assert.equal(
-          options.selfHosted, configObject.lint.selfHosted);
-      });
+      }
+    );
+
+    const options = fakeCommands.lint.firstCall.args[0];
+    // This makes sure that the config object was applied
+    // to the lint command options.
+    assert.equal(
+      options.selfHosted, configObject.lint.selfHosted);
+  });
+
+  it('discovers config files', async () => {
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const configObject = {
+      lint: {
+        selfHosted: true,
+      },
+    };
+    // Instead of loading/parsing a real file, just return an object.
+    const fakeLoadJSConfigFile = sinon.spy(() => {
+      return configObject;
+    });
+
+    await execProgram(
+      ['lint'],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          discoverConfigFiles: async () => ['fake/config.js'],
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      }
+    );
+
+    const options = fakeCommands.lint.firstCall.args[0];
+    // This makes sure that the config object was applied
+    // to the lint command options.
+    assert.equal(
+      options.selfHosted, configObject.lint.selfHosted);
+  });
+
+  it('lets you disable config discovery', async () => {
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+
+    const discoverConfigFiles = sinon.spy(() => Promise.resolve([]));
```

Comment:
Not mandatory, but we could do the same for loadJSConfigFile (defining it as a sinon.spy and assert that it has never been called).

Reply:
The test you commented on is just about disabling config discovery. I think it's enough coverage to just assert that the discovery function is not called. Without that call, there is no way to load a discovered file.

Reply:
yeah, I was thinking more about a possible "future regression" where some new code could try to call loadJSConfigFile without having called discoverConfigFile first, but that's more likely to be just "me being too paranoid" :-P